lib/syslogger/creators.rb

```diff
@@ -4,15 +4,15 @@
 module SysLogger
   module Creators
     def self.unix_dgram_socket(socket_path)
-      proc {
+      Proc.new {
```

Comment:
I'm fine with `raise FooException.new(arg1, arg2)` just like I'm fine with `proc { body }`. But, like I said, I don't really care, I was just going off one style guide I looked at, since they're exactly identical constructs. I think the bigger issue is that I agree with @att14 that it's clearer to just pass a block using block-syntax rather than treating the first argument differently depending on its type, and I don't think the `proc` versus `Proc.new` difference is worth changing if it ends up being the only thing in this PR.